### PR TITLE
Update client.py and README.md

### PR DESCRIPTION
**Pull request by cpeng365**
client.py
1. Added function get_modules in client.py
2. Removed all strip() for response
3. Replaced all tabs with spaces
4. Added private helper function __handshake() that opens the socket, sends the handshake, and closes the socket

README.md
1. Added section for clients

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,6 +105,30 @@ logger = logging.getLogger(__name__)
 logger.info('testing123...')
 ```
 
+## Clients
+Client connects with server.py on host machine to communicate with modules. Here is an example of creating a client.
+```python
+from client import client
+client_1 = client()
```
Comment (mwalsh161) [inline README.md]: let's add `'localhost'` as the argument (even though it is default). On a separate question, is the argument name in client's `__init__` method called `host`. If not, can we change it to that? If it is, let's explicitly say here `host = 'localhost'`.

Comment (mwalsh161) [reply README.md]: Also superrrrrr nitpicky - can we change `client_1` to `myclient` or something similar?

File: README.md [hunk 2]
```diff
@@ -105,6 +105,30 @@ logger = logging.getLogger(__name__)
...
+```
+
+ping(self) can be issued to ping the server to get the client's IP address and binding port.
```
Comment (mwalsh161) [inline README.md]: No need to include the `self` in docs - it is always assumed. Instead let's say "The `ping()` method can be..."

File: README.md [hunk 3]
```diff
@@ -105,6 +105,30 @@ logger = logging.getLogger(__name__)
...
+```python
+print(client_1.ping())
+```
+
+com(self, module, funcname, *args) can be called to communicate with a module and request it to perform a function. Here is an example of requesting "moduleB" to call "fun1" with arguments ['ay', 1, False, None].
```
Comment (mwalsh161) [inline README.md]: same comment here as above, you can omit `self`, especially if you call it a "method"

File: README.md [hunk 4]
```diff
@@ -105,6 +105,30 @@ logger = logging.getLogger(__name__)
...
+```python
+resp = client_1.com('moduleB', 'fun1', 'ay', 1, False, None)
```
Comment (mwalsh161) [inline README.md]: Is this the `moduleB` from above? If so, can you also print the expected output/contents of `resp` so users can compare.

File: client.py [hunk 5]
```diff
@@ -6,9 +6,15 @@
...
+    #  varoius pieces of equipment
+    #  Note, some hwserver operations could conveivably take longer than the
+    #  default timeout used here. If so, obj.connection.Timeout should be
+    #  adjusted appropriately
 
     DEFAULT_IP = 'localhost'
     DEFAULT_PORT = 36577
+    DEFAULT_TIMEOUT = 2
 
     def __init__(self,ip=DEFAULT_IP,port=DEFAULT_PORT):
```
Comment (mwalsh161) [inline client.py]: As per above comment, let's change `ip` to `host` since host can mean host *name* or IP.

File: client.py [hunk 6]
```diff
@@ -44,6 +50,24 @@ def __recv(self,sock,delim='\n',recv_buffer=4096):
                     raise Exception('Server Error: '+msg['response']+'\n'+msg['traceback'])
                 else:
                     return msg['response']
+    
+    def __handshake(self,handshake):
```
Comment (mwalsh161) [inline client.py]: is this used in the `com` method? It could be if you conditionally leave the socket open based on perhaps another input arg.

**Review by mwalsh161 — CHANGES_REQUESTED** [review]
A few minor comments:
 - In readme, if you refer to names of those functions as "methods" you can omit the `self` arg.
 - A few var naming suggestions
 - Compare expected var contents/print output
 - Small change to `__handshake` will make it usable in `com` as well.

**Comment by cpeng365** [comment]
Should all be fixed now! Let me know if there is anything I can change.

**Comment by mwalsh161** [comment]
- Made a few relatively large bug fixes.
 - Decided it would be best to add numpy style docstrings since this file will be used more frequently by users.
- Fixed up logging to be configured by users rather than hardcoded.
- Tested on python 2 and 3.

**Pull request merged**